File: php-advisor-core/src/Payloads/StationPayload.php
<?php

namespace StormGeo\AdvisorCore\Payloads;

class StationPayload extends BasePayload
{
  /**
   * @var string
   */
  public $stationId;

  /**
   * @var string
   */
	public $startDate;

  /**
   * @var string
   */
  public $endDate;

  /**
   * @var array<string>
   */
	public $variables;

  /**
   * @var string
   */
	public $layer;

  /**
   * @param array{stationId:string,startDate:string,endDate:string,variables:array<string>,layer:string} $parameters
   */
  public function __construct($parameters = [])
  {
    parent::__construct(
      ['stationId', 'startDate', 'endDate', 'variables', 'layer'],
      $parameters
    );
  }

  /**
   * @var array
   */
  public function getQueryParams()
  {
    return (array) $this;
  }
}

File: dotnet-advisor-core/Payloads/StationPayload.cs
namespace StormGeo.AdvisorCore.Payloads;

public class StationPayload
{
    public string? StationId { get; set; }
    public string? StartDate { get; set; }
    public string? EndDate { get; set; }
    public string[]? Variables { get; set; }
    public string? Layer { get; set; }

    public string GetQueryParams()
    {
        return new QueryParameterBuilder()
            .AddStationId(StationId)
            .AddStartDate(StartDate)
            .AddEndDate(EndDate)
            .AddVariables(Variables)
            .AddLayer(Layer)
            .Build();
    }
}
